lib/AST/NameLookup.cpp

```diff
@@ -1646,9 +1646,18 @@ bool DeclContext::lookupQualified(Type type,
   // Handle nominal types.
   bool wantProtocolMembers = (options & NL_ProtocolMembers);
   bool wantLookupInAllClasses = false;
-  if (auto nominal = type->getAnyNominal()) {
-    visited.insert(nominal);
-    stack.push_back(nominal);
+  if (auto *NTD = type->getAnyNominal()) {
+    if (NTD->isGenericContext()) {
+      auto *DC = NTD->getDeclContext();
+      // Refuse to return results for situation when
+      // we have a nested type in generic context.
+      if (DC->getAsProtocolOrProtocolExtensionContext() ||
+          (DC->isLocalContext() && DC->isGenericContext()))
+        return false;
```

Comment:
It seems like overkill to completely bail out on returning any results from anything. Should we just "continue" here?

Reply:
Even if we continue this is not going to produce any results because this if else sequence would end up with empty `stack` which I think is the idea since we don't really want to allow any lookup on the types like that, no?

Reply:
Ok, I think I understand what you mean, will push the changes soon...